be/src/vec/common/cow.h

```diff
@@ -92,36 +93,158 @@
   *   to use std::unique_ptr for it somehow.
   */
 template <typename Derived>
-class COW : public boost::intrusive_ref_counter<Derived> {
-private:
+class COW {
+    std::atomic_uint ref_counter;
+
+protected:
+    COW() : ref_counter(0) {}
+
+    COW(COW const&) : ref_counter(0) {}
+
+    COW& operator=(COW const&) {
+        return *this;
+    }
+
+    unsigned int use_count() const {
+        return ref_counter.load();
+    }
+
+    void add_ref() {
+        ++ref_counter;
+    }
+
+    void release_ref() {
```

Comment:
Rethink here may cause mem leak? How boost::intrusive_ref_counter prevent the problem